### DISCUSSION TITLE
What's the difference between PesterConfiguration Exit, Throw, and the default?

### DISCUSSION THREAD

**Discussion by codykonior** [pester/Pester#2133] [Q&A] [answered]
I see in New-PesterConfiguration there's a Run.Throw and a Run.Exit, both disabled by default.

What does the default setting mean?

What do the new settings mean?

There's a line of documentation saying that if Exit and Throw are both used then Throw is preferred, but not what the default behavior is, or what either of them affects.

**Comment by fflaten** [comment, marked as answer]
Default is to print report to screen, set `$LASTEXITCODE` to number of failed tests (this is always done) and proceed with the script that called `Invoke-Pester`.

If any failed tests/blocks:
- `Run.Throw` will make Pester throw an exception at the end with "x tests failed..." so you can wrap `Invoke-Pester` in try/catch.

- `Run.Exit` will exit the process with number of failed tests as exitcode. Typically used to abort a CI-process.
